File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/LifeLink.java
package com.egoal.darkestpixeldungeon.actors.buffs;

import com.egoal.darkestpixeldungeon.messages.Messages;
import com.egoal.darkestpixeldungeon.ui.BuffIndicator;
import com.watabou.utils.Bundle;

/**
 * Created by 93942 on 8/3/2018.
 */

//* check in Char::takeDamage
public class LifeLink extends FlavourBuff {
  {
    type = buffType.POSITIVE;
  }
  
  public int linker = 0;

  private static final String LINKER = "linker";

  @Override
  public void storeInBundle(Bundle bundle) {
    super.storeInBundle(bundle);
    bundle.put(LINKER, linker);

  }

  @Override
  public void restoreFromBundle(Bundle bundle) {
    super.restoreFromBundle(bundle);
    linker = bundle.getInt(LINKER);
  }

  @Override
  public int icon() {
    return BuffIndicator.LIFE_LINK;
  }

  @Override
  public String toString() {
    return Messages.get(this, "name");
  }

  @Override
  public String desc() {
    return Messages.get(this, "desc", dispTurns());
  }
}

File: core/src/main/java/com/egoal/darkestpixeldungeon/actors/buffs/LifeLink.kt
package com.egoal.darkestpixeldungeon.actors.buffs

import com.egoal.darkestpixeldungeon.messages.Messages
import com.egoal.darkestpixeldungeon.ui.BuffIndicator
import com.watabou.utils.Bundle

/**
 * Created by 93942 on 8/3/2018.
 */

//* check in Char::takeDamage
class LifeLink : FlavourBuff() {

    var linker = 0

    init {
        type = Buff.buffType.POSITIVE
    }

    override fun storeInBundle(bundle: Bundle) {
        super.storeInBundle(bundle)
        bundle.put(LINKER, linker)

    }

    override fun restoreFromBundle(bundle: Bundle) {
        super.restoreFromBundle(bundle)
        linker = bundle.getInt(LINKER)
    }

    override fun icon(): Int {
        return BuffIndicator.LIFE_LINK
    }

    override fun toString(): String {
        return Messages.get(this, "name")
    }

    override fun desc(): String {
        return Messages.get(this, "desc", dispTurns())
    }

    companion object {

        private val LINKER = "linker"
    }
}
